src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -233,17 +236,17 @@ class BlobCacheServiceImpl implements BlobCacheService {
      * @return The HTTP URI from the cached layer blob is going to be downloaded
      */
     protected String blobDownloadUri(RoutePath route) {
-        def url = createPresignedGetUrl(blobConfig.storageBucketName, route.targetPath)
-        def downloadUrl
-
-        if( blobConfig.baseUrl ){
-            downloadUrl = url.replaceFirst(/^(https:\/\/[^\/]+)/,blobConfig.baseUrl)
-        }else{
-            downloadUrl = url
+        final bucketPath = StringUtils.pathConcat(blobConfig.storageBucket, route.targetPath)
+        final presignedUrl = createPresignedGetUrl(bucketPath)
+
+        if( blobConfig.baseUrl ) {
+            final p = presignedUrl.indexOf(route.targetPath)
```

Comment:
this index of is not working because of the difference in `:` betweeen presigned and target path

**presigned URL**
https://wave-dev-blob-cache.s3.eu-west-1.amazonaws.com/docker.io/v2/hrma017/dev/blobs/sha256%3A562c6b6eefade9c2c630a62b88ded112cf3f9b589bb546071dea0a819c4f9a75?......

**targetpath**
docker.io/v2/hrma017/dev/blobs/sha256:562c6b6eefade9c2c630a62b88ded112cf3f9b589bb546071dea0a819c4f9a75

Reply:
But according URI spec that's a valid char, there's no need to escape `:`. Fixed

Reply:
I think it depends upon different services
https://stackoverflow.com/a/2053146

Reply:
I think it's ok in the container ecosystem